### ISSUE TITLE
Allow table headers to have different labels for screen readers

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/wc-admin#859]
In several tables we have some columns were we cut the text for spacing reasons but it would make sense to read the entire text for screen readers.

For example, in the _Coupons_ report:

![image](https://user-images.githubusercontent.com/3616980/48561739-3ecb4180-e8b6-11e8-9c23-a383b8d3bc2d.png)

The screen reader text for that column should be _Gross discounted_.

Currently, the `Table` component doesn't allow that but it would be a good addition for accessibility.
